[CI] ArraySearchPhaseResults shardIndex is already set

I've come across various apparently unrelated integration tests with the following error stacktrace:
```
        Caused by:	
        java.lang.AssertionError: shardIndex: 3 is already set	
            at __randomizedtesting.SeedInfo.seed([A373DD6CE0DE449A]:0)	
            at org.elasticsearch.action.search.ArraySearchPhaseResults.consumeResult(ArraySearchPhaseResults.java:43)	
            at org.elasticsearch.action.search.SearchPhaseController$1.consumeResult(SearchPhaseController.java:745)	
            at org.elasticsearch.action.search.AbstractSearchAsyncAction.onShardResult(AbstractSearchAsyncAction.java:471)	
            at org.elasticsearch.action.search.AbstractSearchAsyncAction.access$000(AbstractSearchAsyncAction.java:67)	
            at org.elasticsearch.action.search.AbstractSearchAsyncAction$1.innerOnResponse(AbstractSearchAsyncAction.java:235)
```

Example build scans:
7.x DateHistogramIT https://gradle-enterprise.elastic.co/s/wgy33s2eh6emy
7.x FieldSortIT https://gradle-enterprise.elastic.co/s/nwzskgzmeycmq
7.x MachineLearningLicensingTests https://gradle-enterprise.elastic.co/s/twukyytwyk2vg

It's tedious to find all occurrences of it as this part of the stacktrace is not indexed in build-stats .

--- a/server/src/main/java/org/elasticsearch/search/query/QuerySearchResult.java
+++ b/server/src/main/java/org/elasticsearch/search/query/QuerySearchResult.java
@@ -92,13 +92,11 @@ public final class QuerySearchResult extends SearchPhaseResult {
         this.isNull = isNull;
     }
 
-    private static final QuerySearchResult nullInstance = new QuerySearchResult(true);
-
     /**
      * Returns an instance that contains no response.
      */
     public static QuerySearchResult nullInstance() {
-        return nullInstance;
+        return new QuerySearchResult(true);
     }
 
     /**

